tests/components/tesla_fleet/test_init.py

```diff
@@ -424,3 +424,13 @@ async def test_signing(
     ) as mock_get_private_key:
         await setup_platform(hass, normal_config_entry)
         mock_get_private_key.assert_called_once()
+
+
+async def test_bad_implementation(
+    hass: HomeAssistant,
+    bad_config_entry: MockConfigEntry,
+) -> None:
+    """Test handling of a bad authentication implementation."""
+
+    await setup_platform(hass, bad_config_entry)
+    assert bad_config_entry.state is ConfigEntryState.SETUP_ERROR
```

Comment:
Also assert that reauthentication flow is started?

Reply:
Its not started until the user does so, but I have added that to the test.

Reply:
Thanks my impression is is typically how its tested to make sure its not mixed up with another type of setup error.

Reply:
I see that your first line of `assert any(config_entry.async_get_active_flows(hass, {SOURCE_REAUTH}))` can be sufficient if you don't want to actually "start" it but still verify its active.